### PR TITLE
Switch to Makefile Build System

### PR DESCRIPTION
**Pull request by cfromknecht**
Refactors and extends the `Makefile` created in #962, and modifies travis configuration to use `make travis` instead of the `gotest.sh` script.

- Changes `lntest` to run a local build of `./lnd`, instead of running the version installed in the user's `GOPATH`.
- Adds `Makefile` and `make/testing_flags.mk`, replicates most of the existing behavior such as
    - Dependency management
    - Building and installing lnd
    - Linting and formatting
    - Unit coverage tests
    - Unit race tests
    - Integration testing
    - Travis builds
- Adds additional features:
    - Fetch, glide install, and install btcd
    - Targeted unit/integration testing
    - Flake hunting for unit and integration tests
- Documented the commands in `docs/MAKEFILE.md`
- Updated `docs/INSTALL.md` to use new make directives.
- Removes `gotest.sh`.

 It also fixes an issue caused by tab expansion that resulted in the `icase` argument from being parsed improperly. The forward-looking solution is to remove all indentation from variable definitions, and limit indentation only to build targets.

I've also added a `timeout` param that allows one to specify how long the unit/itest should run before panicking.

### CODE REVIEW

**Comment by halseth** [comment]
Awesome! What about adding a `make flakehunter` command that starts running integration tests in a loop until they fail? 😛

File: Gopkg.lock [hunk 1]
```diff
@@ -359,6 +359,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "0e386a2f0dff9303b03047f7499411379163ed3ca2b4a65161742e1b0fa23ec8"
+  inputs-digest = "a50b1b0760073be728b8e541300b3078f2688d20c61bdd7a0d8115ff7064ae52"
```
Comment (halseth) [inline Gopkg.lock]: Yeah, I was wondering the same 🤔

File: Makefile [hunk 2]
```diff
@@ -1,92 +1,236 @@
...
+	@$(call print, "Building lnd and lncli.")
+	$(GOBUILD) -o lnd $(LDFLAGS) $(PKG)
+	$(GOBUILD) -o lncli $(LDFLAGS) $(PKG)/cmd/lncli
 
 install:
-	@echo "Installing lnd and lncli."
-	go install -v $(LDFLAGS) $(PKG)
-	go install -v $(LDFLAGS) $(PKG)/cmd/lncli
+	@$(call print, "Installing lnd and lncli.")
+	$(CP) ./lnd $(LND_BIN)
```
Comment (Roasbeef) [inline Makefile]: `LND_BIN` and `LNCLI_BIN` aren't defined. IMO this should go back to just using `go install` like before, this'll re-use the binary in place if its already been built.

Comment (cfromknecht) [reply Makefile]: Fixed 👍

**Comment by cfromknecht** [comment]
Pushed up changes, and added the `rpc` directive 🎉

**Pull request merged**